app/assets/javascripts/miq_flash.js

```diff
@@ -46,12 +61,27 @@ function add_flash(msg, level, options) {
   }
 
   $('#flash_msg_div').append(textDiv).show();
+
+  // Check if text is *not* overflowing and options.long_alert is true thus disabling the button.
+  if ((alertDiv[0].scrollWidth <= alertDiv.innerWidth()) && options.long_alert) {
+    $('#alert_expand_button').prop("disabled", true);
+  }
 }
 
 function clearFlash() {
   $('#flash_msg_div').empty();
 }
 
+function expandAlert(params) {
+  if (! params.clicked) {
+    $(".alert").css({height: "auto", 'text-overflow': "initial", "white-space": "normal"});
+    params.clicked = true;
+  } else {
+    $(".alert").css({overflow: "hidden", 'text-overflow': "ellipsis", "white-space": "nowrap"});
```

Comment:
sounds like this is just..

```js
$('.alert').addClass('text-overflow-pf');
```

isn't it? And removeClass for the other case..

Reply:
Also.. you may need to pass in the specific element .. otherwise this will affect all the flash messages, not just the one..